### PR TITLE
Add FP and TN to DeepMIL outputs and configure outputs for multi-class classification

### PR DESCRIPTION
**Pull request by harshita-s**
This PR contains:
- Extension of DeepMIL outputs to contain top/bottom FP and TN slides and tiles
- Extension of DeepMIL outputs to multi-class problems:  top/bottom TP and FN slides and tiles will be generated for each non-zero class
- `test_metrics_utils.py` now includes top/bottom FP and TN slides and tiles

### CODE REVIEW

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (vale-salvatelli) [inline InnerEye/ML/Histopathology/models/deepmil.py:248]: PEP8 (space missing around operator)

Comment (vale-salvatelli) [reply InnerEye/ML/Histopathology/models/deepmil.py:248]: could torch.nn.functional.one_hot be used here for this conversion?

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (vale-salvatelli) [inline InnerEye/ML/Histopathology/models/deepmil.py:258]: shouldn't this be probs_per_class?

Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py:258]: no, probs_perclass is a vector. This remains the same

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py:248]: These are soft probabilities. I am just mapping the probabilities to vector similar to multi-class case.

File: InnerEye/ML/Histopathology/utils/metrics_utils.py
Comment (dccastro) [inline InnerEye/ML/Histopathology/utils/metrics_utils.py:86]: Very minor suggestion to rename `score` to `scores` for clarity that it's now meant to be a collection.

File: Tests/ML/histopathology/utils/test_metrics_utils.py
Comment (dccastro) [inline Tests/ML/histopathology/utils/test_metrics_utils.py:108]: These tests make sense, but I worry there's a risk of the expected values written here and the originals in `test_dict` going out of sync in later changes to this file. Should we create a ticket to later refactor this to access the corresponding values from `test_dict`?

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py:353]: What will be the reference classes for TN[i]  and FP [i]  for multi-class? Currently 0 is the universal negative class

Comment (dccastro) [reply InnerEye/ML/Histopathology/models/deepmil.py:353]: If we choose to include per-class TN/FP, we could frame it as one-vs-all, such that all other classes are negatives.

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py]: Thanks, changed this now

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py]: Thanks for the suggestion, changed now

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py:353]: This could be a deeper discussion. As of now, we have seen 0 as negative. Is it ok if I keep this for now and we could open another PR later

File: InnerEye/ML/Histopathology/utils/metrics_utils.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/utils/metrics_utils.py]: Done

File: Tests/ML/histopathology/utils/test_metrics_utils.py
Comment (harshita-s) [reply Tests/ML/histopathology/utils/test_metrics_utils.py:108]: Thanks. I created a ticket for this

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py]: I have now made change to accommodate binary and multi-class

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (dccastro) [reply InnerEye/ML/Histopathology/models/deepmil.py:354]: If this choice will unblock you for now, we can run with it and add a ticket to revisit later.

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (dccastro) [inline InnerEye/ML/Histopathology/models/deepmil.py:30]: I'd be more in favour of `CLASS_PROBS` (plural), because it will always refer to all classes, but I leave it up to you.

**Review by dccastro — APPROVED** [review]

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py:354]: Added the ticket, thanks!

File: InnerEye/ML/Histopathology/models/deepmil.py
Comment (harshita-s) [reply InnerEye/ML/Histopathology/models/deepmil.py]: Well-spotted, thanks. Changed now

**Review by vale-salvatelli — APPROVED** [review]

**Review by dccastro — APPROVED** [review]

**Pull request merged**
